pkg/certificate/certificate.go

```diff
@@ -115,12 +115,18 @@ func templateAsClientOnly(template *x509.Certificate) *x509.Certificate {
 func templateWithServer(template *x509.Certificate, domain string) *x509.Certificate {
 	template.ExtKeyUsage = append(template.ExtKeyUsage, x509.ExtKeyUsageServerAuth)
 
-	// the specified domain is always the primary
 	template.Subject.CommonName = domain
 
 	// abide by the spec - if CN is an IP, put it in the subjectAltName as well
 	ip := net.ParseIP(domain)
+	if ip == nil {
+		// see if CIDR works
+		ip, _, _ = net.ParseCIDR(domain)
```

Comment:
It can be "" in which case ip will be nil and you end up with a cert that has an empty CommonName field - same as if you didn't call this path at all other than the additional usage ExtKeyUsageServerAuth